### PR TITLE
Simplify some syntax constructions in XamlObjectReader

### PR DESCRIPTION
**Pull request by AndreyAkinshin**

### CODE REVIEW

File: src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Xaml/XamlObjectReader.cs [hunk 1]
```diff
@@ -50,18 +51,12 @@ public XamlObjectReader(object instance, XamlSchemaContext schemaContext)
...
-
-            this.schemaContext = schemaContext;
+            this.schemaContext = schemaContext ?? throw new ArgumentNullException("schemaContext");
             this.settings = settings ?? new XamlObjectReaderSettings();
             this.nodes = new Stack<MarkupInfo>();
             this.currentXamlNode = new XamlNode(XamlNode.InternalNodeType.StartOfStream);
 
-            var context = new SerializerContext(schemaContext, this.settings);
-            context.RootType = instance == null ? null : instance.GetType();
+            var context = new SerializerContext(schemaContext, this.settings) {RootType = instance?.GetType()};
```
Comment (karelz) [inline src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Xaml/XamlObjectReader.cs]: Nit:
```suggestion
            var context = new SerializerContext(schemaContext, this.settings) { RootType = instance?.GetType() };
```

File: src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Xaml/XamlObjectReader.cs [hunk 2]
```diff
@@ -190,11 +185,7 @@ public virtual object Instance
         {
             get
             {
-                if (this.currentXamlNode.NodeType == XamlNodeType.StartObject)
-                {
-                    return this.currentInstance;
-                }
-                return null;
+                return this.currentXamlNode.NodeType == XamlNodeType.StartObject ? this.currentInstance : null;
```
Comment (karelz) [inline src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Xaml/XamlObjectReader.cs]: I'd change it for easier readability (maybe too personal view point?)
```suggestion
                return (this.currentXamlNode.NodeType == XamlNodeType.StartObject) ? this.currentInstance : null;
```

Comment (batzen) [reply src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Xaml/XamlObjectReader.cs]: Adding additional "(" only adds noise, IMHO.

File: src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Xaml/XamlObjectReader.cs [hunk 3]
```diff
@@ -2,6 +2,7 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Linq;
```
[the remark this answers is not in the document]
Comment (danzil) [reply src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Xaml/XamlObjectReader.cs]: I've opened #268 to revert back to not using Linq

**Pull request merged**